integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/search/loading/SearchQueryEntityLoadingScrollingIT.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.mapper.orm.search.loading;
+
+import java.util.List;
+
+import org.hibernate.search.engine.backend.common.DocumentReference;
+import org.hibernate.search.engine.search.query.SearchQuery;
+import org.hibernate.search.engine.search.query.SearchScroll;
+import org.hibernate.search.engine.search.query.SearchScrollResult;
+import org.hibernate.search.util.impl.integrationtest.common.rule.StubSearchWorkBehavior;
+
+public class SearchQueryEntityLoadingScrollingIT extends SearchQueryEntityLoadingBaseIT {
+
+	public SearchQueryEntityLoadingScrollingIT(SingleTypeLoadingModelPrimitives primitives) {
+		super( primitives );
+	}
+
+	@Override
+	protected <T> List<T> getHits(List<String> targetIndexes, SearchQuery<T> query, List<DocumentReference> hitDocumentReferences) {
+		backendMock.expectScrollObjects(
+				targetIndexes,
+				hitDocumentReferences.size(),
+				b -> { }
+		);
+
+		backendMock.expectNextScroll(
+				targetIndexes,
+				StubSearchWorkBehavior.of(
+						hitDocumentReferences.size(),
+						hitDocumentReferences
+				)
+		);
+
+		backendMock.expectCloseScroll( targetIndexes );
+
+		try ( SearchScroll<T> scroll = query.scroll( hitDocumentReferences.size() ) ) {
+			SearchScrollResult<T> next = scroll.next();
+			return next.getHits();
```

Comment:
While this may be useful, I don't think it's enough.

I think we need another test to check that chunk processing works correctly, something like this:

```
	try (Session session = ...) {
		SearchQuery<T> query = ...;
		try ( SearchScroll<T> scroll = query.scroll( hitDocumentReferences.size() ) ) {
			for ( SearchScrollResult<T> next = scroll.next(); next.hasHits(); next = scroll.next() ) {
				// Do something with the hits...
				// (in tests, maybe just update some property)
				session.flush();
				session.clear();
			}
		}
	}
	
	// (in tests, check that changes above had an effect)
```

Maybe you can adapt this test class, maybe you need a separate test class. In any case, we need a test.